### PR TITLE
Replace redsnow with parser

### PR DESCRIPTION
**Pull request by dgmora**
Breaking:
- Before you could parse msons missing definitions. Now nothing gets parsed and you just get the rrors. [Here](https://github.com/moviepilot/lacerda/compare/remove-redsnow?expand=1#diff-310afe66ff82921d4a7a0444f2ce15ebR144). And the [spec](https://github.com/moviepilot/lacerda/compare/remove-redsnow?expand=1#diff-bfa982c6d456c123cbe46c710bf6923eR60). 

TODO:
- Use drafter as a native extension, https://github.com/moviepilot/lounge_lizard/pull/1

### CODE REVIEW

**Comment by dgmora** [comment]
[custom lacerda drafter binding](https://trello.com/c/AuKi4oar/8494-custom-lacerda-drafter-binding)

File: spec/support/contracts/json_schema_test/app/publish.mson [hunk 1]
```diff
@@ -25,7 +25,7 @@ A post that only has a title, but no text body. Weird, right?
 - body: (string) - Body of the post
 - author_id: 2 (number, required) - User id of author
 - primary_tag: (App::Tag, optional) - Property of custom type
-- tags: (array[Tag]) - An array with a custom type
+- tags: (array[App::Tag]) - An array with a custom type
```
Comment (jayniz) [inline spec/support/contracts/json_schema_test/app/publish.mson]: Hmmm.... the way Lacerda has used this feature is:

```markdown
# LocalDefinition
- bar: (number) - This local definition defines an object that is not published by App, i.e. other services can not consume it directly

# App::PublishedDefinition
- id: (number) - This object is prefixed with App:: and other services can consume it
- foo: (LocalDefinition)
```

So I think it's actually useful. What's the reason for this change?

Comment (dgmora) [inline spec/support/contracts/json_schema_test/app/publish.mson]: It seems this is not possible anymore :/. The AST then contains no data structures, just an error saying: "Error code 4: base type 'Tag' is not defined in the document".

Comment (jayniz) [inline spec/support/contracts/json_schema_test/app/publish.mson]: @dgmora I think there was a misunderstanding. What you did:

```markdown
# App:Object1
- id: (number)

# App::Object2
- foo: (array[App::Object1])
```

and not

```markdown
# App::Object2
- foo: (array[Object1])
```

That is totally correct, and if this worked before than that was a bug. 

What I thought you meant, because I didn't take a proper look, is: the following is not valid, only `# App::LocalObject` is supported, but not:

```markdown
# LocalObject
- id: (number)
```

So all is good :)

File: lib/lacerda/conversion/data_structure.rb [hunk 2]
```diff
@@ -48,49 +48,54 @@ def to_json
       private
 
       def add_description_to_json_schema
-        return unless @data['sections']
-        description = @data['sections'].select{|d| d['class'] == 'blockDescription' }.first
+        # Should we check here if content is an array as well?
```
Comment (jayniz) [inline lib/lacerda/conversion/data_structure.rb]: Que vols dir?

**Comment by jayniz** [comment]
> All checks have failed
> 2 failing checks

ahem, who approved this 🙈

File: lib/lacerda/conversion/data_structure.rb (hunk 2 above)
Comment (dgmora) [inline lib/lacerda/conversion/data_structure.rb]: Good question!
